contrib/platform/src/com/sun/jna/platform/win32/Guid.java

```diff
@@ -466,16 +468,14 @@ public CLSID(GUID guid) {
 
     /**
      * The Class REFIID.
-     *
-     * @author Tobias Wolf, wolf.tobias@gmx.net
      */
-    public class REFIID extends IID {
+    public class REFIID extends ByReference {
```

Comment:
My concern is that this has exposed a latent bug and we’d only be masking it.

Your definition is otherwise sane and certainly not misleading (except that it varies from the recommended method of establishing “struct” vs “struct *”).

Give me a few days to take a look at this (I’ve been totally swamped this week paying face time to visiting dignitaries).

T.

> On Jan 13, 2016, at 11:03 AM, matthiasblaesing <notifications@github.com> wrote:
> 
> In contrib/platform/src/com/sun/jna/platform/win32/Guid.java:
> 
> >       */
> > -    public class REFIID extends IID {
> > +    public class REFIID extends ByReference {
> 
> My reasoning for the com.sun.jna.ptr.ByReference approach was that it is closest to the underlying definition. It is clear by that no magic handling is expected from JNA, just passing the pointer.
> 
> If this is strictly out of line - I only see the Option to make the members of the GUID structure final so that when passed in they are only read and in general GUID must be interpreted as a value type.
> 
> —
> Reply to this email directly or view it on GitHub.
>